src/message/messagesActions.js

```diff
@@ -47,5 +50,23 @@ export const messageLinkPress = (href: string) => (dispatch: Dispatch, getState:
     dispatch(doNarrow(narrow, anchor));
     return;
   }
-  openLink(getFullUrl(href, auth.realm));
+  const url = getFullUrl(href, auth.realm);
+
+  // We don't want to generate temporary URLs to external websites.
+  if (!url.includes(auth.realm)) {
+    openLink(url);
+    return;
+  }
+
+  let tempUrl: string;
+  try {
+    tempUrl = getFullUrl((await getTemporaryFileUrl(auth, href)).url, auth.realm);
+  } catch (err) {
+    // If we failed to generate the temporary URL, we assume it's because the
+    // server doesn't suport it yet, and fallback to use the regular file
+    // URL.
+    openLink(url);
+    return;
+  }
+  openLink(tempUrl);
```

Comment:
This logic is kind of tricky, especially around the try/catch.

A couple of things that I think would help:
* Every control-flow path out of this code ends up calling `openLink` at the end. Make that explicit by having just one `openLink` call. Have a helper function that returns the URL we'll pass to `openLink`. That function will contain some control flow.

  This also makes an opportunity to give that helper function a meaningful name, which I think will also add clarity.

* Have the `try` block be as small as possible. As is, we'd end up in the `catch` if there's somehow an exception thrown from `getFullUrl`. That shouldn't happen, but ruling that out requires going and reading more code. Instead that call can be after the whole try/catch.